ocaml/sdk-gen/component-test/jsonrpc-client/go/main_test.go

```diff
@@ -50,3 +59,881 @@ func TestLoginSuccess(t *testing.T) {
+	var result xenapi.VMRecord
+	var expectedResult = xenapi.VMRecord{
+		UUID:             "c5db1298-3e97-53bb-4f97-bd039ca8c217",
+		IsATemplate:      false,
+		MemoryStaticMax:  8589934592,
+		MemoryDynamicMax: 8589934592,
+		MemoryDynamicMin: 8589934592,
+		MemoryStaticMin:  4294967296,
+		VTPMs: []xenapi.VTPMRef{
+			"OpaqueRef:60f5a19e-773b-f831-c9c2-35f07b632603",
+		},
+		HVMBootParams: map[string]string{
+			"firmware": "uefi",
+			"order":    "cd",
+		},
+		HVMShadowMultiplier: 1.0,
+		SnapshotTime:        time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC),
+	}
+	result, err = xenapi.VM.GetRecord(session, vmRef)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMGetDataSources(t *testing.T) {
+	session, err := getSession("test_id7")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"
+	var result []xenapi.DataSourceRecord
+	var expectedResult = []xenapi.DataSourceRecord{
+		{
+			NameLabel:       "memory",
+			NameDescription: "Memory currently allocated to VM",
+			Enabled:         true,
+			Standard:        true,
+			Units:           "B",
+			Min:             0.0,
+			Max:             math.Inf(1),
+			Value:           0.0,
+		},
+	}
+	result, err = xenapi.VM.GetDataSources(session, vmRef)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMSnapshot(t *testing.T) {
+	session, err := getSession("test_id8")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"
+	var expectedResult xenapi.VMRef = "OpaqueRef:snapshot-0bf0-30ff-804f-5f0ee4bbdd13"
+	var result xenapi.VMRef
+	ignoreVdis := []xenapi.VDIRef{
+		"OpaqueRef:cf90294c-3881-bf1a-55fc-f2eb6c0cbd8a",
+	}
+	result, err = xenapi.VM.Snapshot(session, vmRef, "snapshot0", ignoreVdis)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if result != expectedResult {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMGetSnapshotTime(t *testing.T) {
+	session, err := getSession("test_id9")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	var vmRef xenapi.VMRef = "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"
+	var result time.Time
+	var expectedResult time.Time = time.Date(2024, time.April, 20, 12, 0, 0, 0, time.UTC)
```

Comment:
I think for this test case, we want to try other time format besides the exists time format in other test cases, eg. time.RFC3339, "20060102T15:04:05Z", "20060102T15:04:05".